### ISSUE TITLE
Danger House: Platform camera panning issues.

### ISSUE THREAD

**Issue by DDR0** [frogatto/frogatto#731]
flipping the switch then plays the animation showing you it activated the moving platform - that animation seems to bugout for me
the bugging out was when I turned it off the platform teleported to the top, so it looked weird, guess this is now a suggestion to.. not show that?
e:  also you can just skip activating the lever if you wall-jump slightly well.. might be intentional, thought to mention it just in case 

![image](https://github.com/user-attachments/assets/abd1d931-4ff7-4729-acd3-2008f2232480)

[Reported](https://discord.com/channels/225816341737766912/920412269718953994/1307806206794137612) by Luka116.

**Comment by rkettering** [comment]
Yeah, I'm not a fan of the platform teleporting, but it's based on our **ancient** platform code that's purely stateless.   Despite having a pretty scorched-earth attitude towards introducing state, this is a case where doing so (again, in the most minimalistic way possible) is the appropriate fix.

It's probably fine to do it in the prototype; it's not used by _much_, but we may as well "fix it for all of these" at once.  I think the only platforms that still use this ancient code are the seaside ones.
